### PR TITLE
Fix technical weakness in event table managers

### PR DESCRIPTION
**Pull request by Bekkalizer**
The event table manager was counting both the event and enrollment tables as its own. At the same time the enrollment table manager did not return any tables in the getExistingDatabaseTables() function.

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/partition/JdbcPartitionManager.java [hunk 1]
```diff
@@ -78,6 +78,17 @@
...
+        return getTablesWithNameLikeness( AnalyticsTableType.EVENT.getTableName() );
+    }
+
+    @Override
+    public Set<String> getEnrollmentAnalyticsPartitions()
+    {
+        return getTablesWithNameLikeness( AnalyticsTableType.ENROLLMENT.getTableName() );
+    }
+
+    private Set<String> getTablesWithNameLikeness( String nameLikeness )
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/partition/JdbcPartitionManager.java]: Rename to:

```
private Set<String> getAnalyticsPartitions( AnalyticsTableType tableType )
```

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/partition/JdbcPartitionManager.java [hunk 2]
```diff
@@ -48,52 +50,30 @@
 {
     private static final Log log = LogFactory.getLog( JdbcPartitionManager.class );
 
-    private Set<String> analyticsPartitions = null;
-    private Set<String> analyticsEventPartitions = null;
-
-    //TODO separate method for enrollment partitions ?
+    private Map<AnalyticsTableType, Set<String>> analyticsPartitions = 
+        new HashMap<AnalyticsTableType, Set<String>>();
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/partition/JdbcPartitionManager.java]: use `new HashMap<>();`

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcAnalyticsTableManager.java [hunk 3]
```diff
@@ -128,7 +128,7 @@ public AnalyticsTableType getAnalyticsTableType()
     @Override
     public Set<String> getExistingDatabaseTables()
     {
-        return partitionManager.getDataValueAnalyticsPartitions();
+        return partitionManager.getAnalyticsPartitions( AnalyticsTableType.DATA_VALUE );
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcAnalyticsTableManager.java]: `return partitionManager.getAnalyticsPartitions( getAnalyticsTableType() );`

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcEnrollmentAnalyticsTableManager.java [hunk 4]
```diff
@@ -98,7 +98,7 @@ public AnalyticsTableType getAnalyticsTableType()
     @Override
     public Set<String> getExistingDatabaseTables()
     {
-        return new HashSet<>(); //TODO this must be implemented properly
+        return partitionManager.getAnalyticsPartitions( AnalyticsTableType.ENROLLMENT );
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcEnrollmentAnalyticsTableManager.java]: `return partitionManager.getAnalyticsPartitions( getAnalyticsTableType() );`

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcEventAnalyticsTableManager.java [hunk 5]
```diff
@@ -120,7 +120,7 @@ public AnalyticsTableType getAnalyticsTableType()
     @Override
     public Set<String> getExistingDatabaseTables()
     {
-        return partitionManager.getEventAnalyticsPartitions();
+        return partitionManager.getAnalyticsPartitions( AnalyticsTableType.EVENT );
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcEventAnalyticsTableManager.java]: `return partitionManager.getAnalyticsPartitions( getAnalyticsTableType() );`

**Comment by Bekkalizer** [comment]
Put the getAnalyticsPartitions methods in an abstract base class.

**Pull request merged**